civis/base.py

```diff
@@ -135,7 +137,12 @@ def _make_request(self, method, path=None, params=None, data=None,
 
         return response
 
-    def _call_api(self, method, path=None, params=None, data=None, **kwargs):
+    def _call_api(self, method, path=None, params=None, data=None,
+                  deprecation_warning=None, **kwargs):
+        if deprecation_warning:
+            # stacklevel=3 to point to the call just outside civis-python
+            warnings.warn(deprecation_warning, FutureWarning, stacklevel=3)
```

Comment:
To see this action, one of the deprecated endpoint method is `client.tables.post_refresh` (as of May 2024). If we run the following code:

```python
import civis
from civis.base import CivisAPIError
from civis.resources import API_SPEC_PATH

fake_table_id = 123

# Use the local API spec to avoid making actual Civis API calls.
client = civis.APIClient(local_api_spec=API_SPEC_PATH)

try:
    client.tables.post_refresh(fake_table_id)
except CivisAPIError as e:
    if e.status_code == 404:
        # The table {fake_table_id} doesn't exist.
        # This 404 error is irrelevant for our test of FutureWarning
        # for a deprecated endpoint method.
        pass
    else:
        raise
```

we get this warning that points to where the line of code with `client.tables.post_refresh` is:

```
<some-local-path-and-line-number>: FutureWarning: Warning: The tables/:id/refresh endpoint is deprecated. Please use tables/scan from now on.
  client.tables.post_refresh(fake_table_id)
```